libs/sdk-core/src/greenlight/node_api.rs

```diff
@@ -2114,16 +2163,29 @@ impl TryFrom<cln::ListpaysPays> for Payment {
             .as_ref()
             .ok_or(InvoiceError::generic("No bolt11 invoice"))
             .and_then(|b| parse_invoice(b));
-        let payment_amount = payment
-            .amount_msat
-            .clone()
-            .map(|a| a.msat)
-            .unwrap_or_default();
         let payment_amount_sent = payment
             .amount_sent_msat
             .clone()
             .map(|a| a.msat)
             .unwrap_or_default();
+
+        // For trampoline payments the amount_msat doesn't match the actual
+        // amount. If it's a trampoline payment, take the amount from the label.
+        let (payment_amount, client_label) = serde_json::from_str::<PaymentLabel>(payment.label())
```

Comment:
Do we have the bolt11 we paid in case of trampoline? If so perhaps better to take the amount from there? If we don't perhaps it is better to persist it as part of the label? I think from UX perspective the user needs the paid invoice.

Reply:
The invoice is there, but possibly it's a zero amount invoice (which we support). Having the amount in the label handles both cases.

Reply:
Right. I think we have a mechanism that adds the amount (in case of zero invoice) to the extrenal_info (persist_pending_payment).
But I am not against of having that in the label which I think is better.